src/view/util/view_utils.py

```diff
@@ -0,0 +1,36 @@
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def get_widget_by_accessor_function(widget, target_accessor):
+    """
+    This method searches for a widget using the provided target accessor function. It first checks if the widget is a nested widget by calling
+    the _process_nested_widgets function recursively. If a nested widget is found, it is returned. If no nested widget is found, it checks if
+    the widget has a 'params' attribute and if the 'accessor' key is present in widget.params with a value matching the target_accessor.
+    If a match is found, the widget is returned. If no widget is found matching the target_accessor, None is returned.
+
+    :param widget: The widget from which to search for the target widget.
+    :param target_accessor: The accessor function used to identify the target widget.
+    :return: The target widget if found, otherwise None.
+    """
+    nested_widget_result = _process_nested_widgets(widget, target_accessor)
+    if nested_widget_result is not None:
+        return nested_widget_result
+    if hasattr(widget, 'params') and "accessor" in widget.params and widget.params["accessor"] == target_accessor:
```

Comment:
Wenn du den check vor dem nested machst kannst du danach einfach den return von _process_nested_widget weiterreichen ohne nochmal einen null check zu machen

Reply:
Die Funktion _process_nested_widgets wird für jedes untergeordnete Widget rekursiv aufgerufen und gibt das erste Widget zurück, das die Bedingung erfüllt (result is not None). Wenn jedoch kein passendes Widget unter den untergeordneten Widgets gefunden wird, gibt _process_nested_widgets None zurück.
Der Check if nested_widget_result is not None: in get_widget_by_accessor_function stellt fest, ob ein passendes Widget unter den untergeordneten Widgets gefunden wurde. Wenn das der Fall ist (d.h., nested_widget_result ist nicht None), wird das gefundene Widget sofort zurückgegeben und die Suche beendet.
Wenn man den Check entfernt, würde die Funktion get_widget_by_accessor_function das aktuelle Widget auf Übereinstimmung überprüfen, auch wenn bereits ein passendes Widget in den untergeordneten Widgets gefunden wurde. Dies kann dazu führen, dass ein falsches Widget zurückgegeben oder unnötige Checks durchgeführt werden.

Reply:
Mein Punkt war dass, wenn du zuerst prüfst ob das aktuelle widget matched, dann kannst du das einfach zurückgeben. Danach erst in die Rekusion gehen.
Damit sparst du dir ein if.
```suggestion
    if hasattr(widget, 'params') and "accessor" in widget.params and widget.params["accessor"] == target_accessor:
       return widget
    return _process_nested_widgets(widget, target_accessor)

```